lib/ecto/query/inspect.ex

```diff
@@ -27,7 +27,22 @@ defimpl Inspect, for: Ecto.Query do
         string
     end)
 
-    surround_many("#Ecto.Query<", list, ">", opts, fn str, _ -> str end)
+    result = surround_many("#Ecto.Query<", list, ">", opts, fn str, _ -> str end)
+
+    case query.with_ctes do
+      %WithExpr{recursive: recursive, queries: [_] = queries} ->
+        with_ctes =
+          Enum.map(queries, fn {name, query} ->
+            cte = query |> __MODULE__.inspect(opts)
```

Comment:
```suggestion
            cte = inspect(query, opts)
```

Reply:
`__MODULE__` in necessary because:

```
** (CompileError) lib/ecto/query/inspect.ex:28: imported Kernel.inspect/2 conflicts with local function
```